### DISCUSSION TITLE
Max parallel has undesirable blocking behavior

### DISCUSSION THREAD

**Discussion by chingc** [community/community#26042] [Actions]
<p>The <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymax-parallel" rel="nofollow noopener">max-parallel</a> setting seems to have blocking behavior.</p>
<p>For example, if I have 20 total jobs and max-parallel is 5, GitHub Actions will start 5 jobs but will not start more until all 5 are complete.</p>
<p>It would be ideal if new jobs are started immediately as old jobs are finished, maintaining the max-parallel value of 5 at all times.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/chingc">@chingc</a>,</p>
<blockquote>
<p>For example, if I have 20 total jobs and max-parallel is 5, GitHub Actions will start 5 jobs but will not start more until all 5 are complete.</p>
</blockquote>
<p>I also tested this with GitHub-hosted runners and self-hosted runners, however I did not reproduce the issue likes as you mentioned.<br>
In my test, after one job in a batch was completed, a queued job would start, it did not wait other jobs in that batch to be completed, although there may be some delay to start the queued job after the previous job completed.</p>
<p>I noticed this description in the docs:</p>
<blockquote>
<p>By default, GitHub will maximize the number of jobs run in parallel <strong>depending on the available runners on GitHub-hosted virtual machines</strong>.</p>
</blockquote>
<p>When using GitHub-hosted runners, there may be some usage limits.<br>
If you are using GitHub-hosted runners in your matrix, you can try to use some self-hosted runners to see if the issue still exists.</p>

**Comment by chingc** [comment]
<p><a class="mention" href="/brightran">@brightran</a> I played with it some more last night and it looks like you are correct and I was mistaken.   Thanks for the note about usage limits.</p>
